src/main/java/de/tum/in/www1/artemis/web/rest/metis/conversation/ConversationResource.java

```diff
@@ -124,6 +129,39 @@ public ResponseEntity<Boolean> hasUnreadMessages(@PathVariable Long courseId) {
         return ResponseEntity.ok(conversationService.userHasUnreadMessages(courseId, requestingUser));
     }
 
+    /**
+     * GET /api/courses/:courseId/code-of-conduct-agreement : Checks if the user agrees to the code of conduct
+     *
+     * @param courseId the course's ID
+     * @return ResponseEntity with status 200 (Ok) and body is true if the user agreed to the course's code of conduct
+     */
+    @GetMapping("/{courseId}/code-of-conduct-agreement")
+    @EnforceAtLeastStudent
+    public ResponseEntity<Boolean> isCodeOfConductAccepted(@PathVariable Long courseId) {
+        checkMessagingEnabledElseThrow(courseId);
+        var course = courseRepository.findByIdElseThrow(courseId);
+        var requestingUser = userRepository.getUserWithGroupsAndAuthorities();
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, courseRepository.findByIdElseThrow(courseId), requestingUser);
+        return ResponseEntity.ok(courseCodeOfConductAgreementService.fetchUserAgreesToCodeOfConductInCourse(requestingUser, course));
+    }
+
+    /**
+     * POST /api/courses/:courseId/code-of-conduct-agreement : Accept the course's code of conduct
+     *
+     * @param courseId the course's ID
+     * @return ResponseEntity with status 200 (Ok) and body is true if the user agreed to the code of conduct
+     */
+    @PostMapping("/{courseId}/code-of-conduct-agreement")
+    @EnforceAtLeastStudent
+    public ResponseEntity<Boolean> acceptCodeOfConduct(@PathVariable Long courseId) {
+        checkMessagingEnabledElseThrow(courseId);
+        var course = courseRepository.findByIdElseThrow(courseId);
+        var requestingUser = userRepository.getUserWithGroupsAndAuthorities();
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, courseRepository.findByIdElseThrow(courseId), requestingUser);
+        courseCodeOfConductAgreementService.setUserAgreesToCodeOfConductInCourse(requestingUser, course);
+        return ResponseEntity.ok(courseCodeOfConductAgreementService.fetchUserAgreesToCodeOfConductInCourse(requestingUser, course));
```

Comment:
Why do you have to fetch it again from the data base? And why do you need the returned value at all, shouldn't it be always true?